gateway/cli/src/main.rs

```diff
@@ -101,93 +105,88 @@ async fn main() {
             println!("version: {}", env!("GIT_HASH"));
         }
         Commands::Info => {
-            call(
-                source_password(cli.rpcpassword),
-                cli.url,
-                String::from("/info"),
-                (),
-            )
-            .await;
+            let response = client
+                .get_info(source_password(cli.rpcpassword))
+                .await
+                .expect("Failed to get info");
+
+            print_response(response).await;
         }
         Commands::Balance { federation_id } => {
-            call(
-                source_password(cli.rpcpassword),
-                cli.url,
-                String::from("/balance"),
-                BalancePayload { federation_id },
-            )
-            .await;
+            let response = client
+                .get_balance(
+                    source_password(cli.rpcpassword),
+                    BalancePayload { federation_id },
```

Comment:
Questionable if we still need the payload struct here since we got specialized functions for each RPC method  now

Reply:
Opportunity for future polish